src/test/playwright/support/pageobjects/exercises/programming/RepositoryPage.ts

```diff
@@ -13,16 +14,20 @@ export class RepositoryPage {
     }
 
     async checkCommitHistory(commits: ExerciseCommit[]) {
+        // Reverse commit history as the latest commit in commit history table is at the bottom
+        commits = commits.reverse();
         const commitHistory = this.page.locator('.card-body', { hasText: 'Commit History' });
 
         if (commits) {
             const commitCount = commits.length;
-            for (let index = 0; index < commitCount; index++) {
+            for (let index = commitCount - 1; index > 0; index--) {
                 const commit = commits[index];
                 const commitRow = commitHistory.locator('tbody').locator('tr').nth(index);
                 await expect(commitRow.locator('td').getByText(commit.message)).toBeVisible();
+
                 if (commit.result) {
-                    await expect(commitRow.locator('#result-score', { hasText: commit.result })).toBeVisible();
+                    const commitResult = commitRow.locator('#result-score', { hasText: commit.result });
+                    await Commands.reloadUntilFound(this.page, commitResult, 2000, 60000);
```

Comment:
The magic numbers for the timeout and interval could be defined in constants at the start of the class to improve readability